extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/base/ModeledDirectoryObjectService.java

```diff
@@ -97,9 +97,12 @@
      *
      * @return
      *     An object which is backed by the given model object.
+     *
+     * @throws GuacamoleException
+     *     If the object instance cannot be created.
      */
     protected abstract InternalType getObjectInstance(ModeledAuthenticatedUser currentUser,
-            ModelType model);
+            ModelType model) throws GuacamoleException;
```

Comment:
Ah - because `getObjectInstance()` now needs to call `hasObjectPermission()`, at least in the user case. See:

https://github.com/mike-jumper/incubator-guacamole-client/blob/232935fe2cd81077c81bed6079707e3a41ceb9fb/extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/user/UserService.java#L161-L162